src/components/NoteList.js

```diff
@@ -17,16 +19,29 @@ NoteItem.defaultProps = {
   title: PropTypes.string,
 };
 
-const Notes = props => (
-  <ul>{props.notes.map(note => <NoteItem key={note.id} {...note} />)}</ul>
-);
+class NoteList extends Component {
```

Comment:
Now the fetchNotes() is effectively an API call. It needs to happen before the component fully loads otherwise the `this.props.notes` will be undefined. You have to have a class to use `componentDidMount()`. This ensures that the fetchNotes gets called in the right order.